agent/taskresource/credentialspec/credentialspec_windows.go

```diff
@@ -376,3 +483,128 @@ func (cs *CredentialSpecResource) MarshallPlatformSpecificFields(credentialSpecR
 func (cs *CredentialSpecResource) UnmarshallPlatformSpecificFields(credentialSpecResourceJSON CredentialSpecResourceJSON) {
 	return
 }
+
+func setTaskExecutionCredentialsRegKeys(taskCredentials credentials.IAMRoleCredentials, taskArn string) error {
+	if taskCredentials == (credentials.IAMRoleCredentials{}) {
+		err := errors.New("Unable to find execution role credentials while setting registry key for task " + taskArn)
+		return err
+	}
+
+	taskRegistryKey, _, err := registry.CreateKey(registry.LOCAL_MACHINE, ecsCcgPluginRegistryKeyRoot+"\\"+taskArn, registry.WRITE)
+	if err != nil {
+		seelog.Errorf("Error creating registry key root %s for task %s: %s", ecsCcgPluginRegistryKeyRoot, taskArn, err)
+		return err
+	}
+	defer taskRegistryKey.Close()
+
+	err = taskRegistryKey.SetStringValue("AKID", taskCredentials.AccessKeyID)
+	if err != nil {
+		seelog.Errorf("Error creating AKID child value for task %s:%s", taskArn, err)
+		return err
+	}
+	err = taskRegistryKey.SetStringValue("SKID", taskCredentials.SecretAccessKey)
+	if err != nil {
+		seelog.Errorf("Error creating SKID child value for task %s:%s", taskArn, err)
+		return err
+	}
+	err = taskRegistryKey.SetStringValue("SESSIONTOKEN", taskCredentials.SessionToken)
+	if err != nil {
+		seelog.Errorf("Error creating SESSIONTOKEN child value for task %s:%s", taskArn, err)
+		return err
+	}
+
+	return nil
+}
+
+func handleNonFileDomainlessGMSACredSpec(originalCredSpec, localCredSpecFilePath, taskARN string) error {
+	// Exit early for non domainless gMSA cred specs
+	if !strings.HasPrefix(originalCredSpec, "credentialspecdomainless:") {
+		return nil
+	}
+
+	err := readWriteDomainlessCredentialSpecImpl(localCredSpecFilePath, localCredSpecFilePath, taskARN)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func readWriteDomainlessCredentialSpec(filePath, outFilePath, taskARN string) error {
+	credSpec, err := readCredentialSpecImpl(filePath)
+	if err != nil {
+		return err
+	}
+	err = writeCredentialSpecImpl(credSpec, outFilePath, taskARN)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func readCredentialSpec(filePath string) (map[string]interface{}, error) {
+	byteResult, err := osReadFileImpl(filePath)
+	if err != nil {
+		return nil, err
+	}
+	var credSpec map[string]interface{}
+	err = json.Unmarshal(byteResult, &credSpec)
+	if err != nil {
+		return nil, err
+	}
+	return credSpec, nil
+}
+
+func writeCredentialSpec(credSpec map[string]interface{}, outFilePath string, taskARN string) error {
+	activeDirectoryConfigUntyped, ok := credSpec["ActiveDirectoryConfig"]
```

Comment:
That is the approach that Linux domainless gMSA does. Windows cannot use that approach.


The reason that Linux is able to do that approach is because they are in charge of defining their credential specs. Linux gMSA is an entirely hack around design and not natively supported, so Linux can control the fields in their credential spec. 

Windows gMSA is a native concept supported by Microsoft and the Windows Operating system. Unfortunately, Microsoft has not published official documentation on the structure of the credential specs, and we do not want to assume beyond the bare minimum. If we were to define a struct, we could end up ignoring other fields and breaking functionality in the long term future. I have also discussed this with Justin from our team who agreed with the approach.

Reply:
Okay I understand that we want to capture all the data while parsing the fields that we care about. Let's add comments explaining this so that future readers know why the parsing is done manually.

Reply:
done